main.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    counter = 0
+    while counter < size
+      yield(self[counter])
+      counter += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    output = []
+    my_each { |ele| output << ele if yield(ele) }
+    output
+  end
+
+  def my_all?
+    output = true
+    my_each do |ele|
+      output = false if yield(ele) != true
+    end
+    output
+  end
+
+  def my_any?
+    output = false
+    my_each do |ele|
+      output = true if yield(ele) == true
+    end
+    output
+  end
+
+  def my_none?
+    output = true
+    my_each do |ele|
+      output = false if yield(ele) == true
+    end
+    puts output
+  end
+
+  def my_count(number = nil)
```

Comment:
#### `REQUIRED`

### :bug: Let's add a case for when a block is not given. Compare with the built-in method. Learn more [here](https://ruby-doc.org/core-2.6.3/Enumerable.html#method-i-count)

#### 1. When `arg` is not a `block`

```ruby
irb(main):001:0> require_relative 'main'
=> true
irb(main):002:0> [1,2,3,4,5].count(3)
=> 1
irb(main):003:0> [1,2,3,4,5].my_count(3)
=> 0
```